### PR TITLE
removal: contributors script

### PR DESCRIPTION
**Pull request by consideRatio**
Running this script typically takes more than an hour, it takes quite a
while to make all the required requests, but it can also require waiting
for refreshed GitHub API quota.

I'd like delete it or make adjustments to how it is used currently, this
relates to:
- running it is a cumbersome step as part of the release process
- running it takes 30-120 minutes to execute
- it includes jupyterhub, kubespawner, oauthenticator as well
- it requires instructions to use and perserverence to use successfully
  - specific python libraries needs to be installed
  - a GitHub API key needs to be configured
  - dates since last use needs to be adjusted
- PR authors are already credited separately

Ping @yuvipanda as the author of this script and @choldgraf @bitnik as [contributor to this script](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/commits/master/tools/contributors.py).

### CODE REVIEW

**Comment by yuvipanda** [comment]
Thanks for opening this, @consideRatio!

I think it's important to credit *all* the people who have contributed in any form to z2jh, which includes all the subprojects this relies on & all the ways they have done so. To that end, I'd like to keep this script. If the speed is the issue, I'm sure we can find ways to make it faster - or replace it with another script.

What do you think?

**Comment by consideRatio** [comment]
@yuvipanda my biggest concern is that it slows down and makes the release process more complicated. If we would define a way to publish this separately from the release itself, make it run at least fast enough to not need to await GitHub API quota or similar, that would be sufficient to make me happy.

I think related work to do regards the changelog, that contains too much at the same time. It currently acts to provide release notes, the actual changelog, upgrade instructions, contributor credits - all in one place.

**Comment by manics** [comment]
I think there are two issues here:
- difficulty in running the script
- overall burden on the person, and it is often _one_ person, making the release (which is not limited to this script, it also includes all the things leading up to the release).

Do you think the latter is worth a separate team discussion?

**Comment by choldgraf** [comment]
Is this something that we could use `github-activity` for? It can return markdown lists of contributors between dates as well. E.g., for me, running:

```
github-activity jupyterhub/zero-to-jupyterhub-k8s -s 0.7.0 -u 0.8.0
```

takes about 5-10 seconds, and results in the list at the bottom of this comment.

[see here for how github-activity defines "contributor"](https://github-activity.readthedocs.io/en/latest/#how-does-this-tool-define-contributions-in-the-reports)
---
## Contributors to this release
([GitHub contributors page for this release](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/graphs/contributors?from=2018-09-03&to=2019-02-21&type=c))

[@01100010011001010110010101110000](https://github.com/search?q=repo%3Ajupyterhub%2Fzero-to-jupyterhub-k8s+involves%3A01100010011001010110010101110000+updated%3A2018-09-03..2019-02-21&type=Issues) | [@ablekh](https://github.com/search?q=repo%3Ajupyterhub%2Fzero-to-jupyterhub-k8s+involves%3Aablekh+updated%3A2018-09-03..2019-02-21&type=Issues) | [@Aisuko](https://github.com/search?q=repo%3Ajupyterhub%2Fzero-to-jupyterhub-k8s+involves%3AAisuko+updated%3A2018-09-03..2019-02-21&type=Issues) | [@alanjcastonguay](https://github.com/search?q=repo%3Ajupyterhub%2Fzero-to-jupyterhub-k8s+involves%3Aalanjcastonguay+updated%3A2018-09-03..2019-02-21&type=Issues) | [@amanda-tan](https://github.com/search?q=repo%3Ajupyterhub%2Fzero-to-jupyterhub-k8s+involves%3Aamanda-tan+updated%3A2018-09-03..2019-02-21&type=Issues) | [@arokem](https://github.com/search?q=repo%3Ajupyterhub%2Fzero-to-jupyterhub-k8s+involves%3Aarokem+updated%3A2018-09-03..2019-02-21&type=Issues) | [@azbones](https://github.com/search?q=repo%3Ajupyterhub%2Fzero-to-jupyterhub-k8s+involves%3Aazbones+updated%3A2018-09-03..2019-02-21&type=Issues) | [@betatim](https://github.com/search?q=repo%3Ajupyterhub%2Fzero-to-jupyterhub-k8s+involves%3Abetatim+updated%3A2018-09-03..2019-02-21&type=Issues) | [@bitnik](https://github.com/search?q=repo%3Ajupyterhub%2Fzero-to-jupyterhub-k8s+involves%3Abitnik+updated%3A2018-09-03..2019-02-21&type=Issues) | [@boileaum](https://github.com/search?q=repo%3Ajupyterhub%2Fzero-to-jupyterhub-k8s+involves%3Aboileaum+updated%3A2018-09-03..2019-02-21&type=Issues) | [@BrianVanEtten](https://github.com/search?q=repo%3Ajupyterhub%2Fzero-to-jupyterhub-k8s+involves%3ABrianVanEtten+updated%3A2018-09-03..2019-02-21&type=Issues) | [@cam72cam](https://github.com/search?q=repo%3Ajupyterhub%2Fzero-to-jupyterhub-k8s+involves%3Acam72cam+updated%3A2018-09-03..2019-02-21&type=Issues) | [@camilo-nunez](https://github.com/search?q=repo%3Ajupyterhub%2Fzero-to-jupyterhub-k8s+involves%3Acamilo-nunez+updated%3A2018-09-03..2019-02-21&type=Issues) | [@canhtran](https://github.com/search?q=repo%3Ajupyterhub%2Fzero-to-jupyterhub-k8s+involves%3Acanhtran+updated%3A2018-09-03..2019-02-21&type=Issues) | [@cbjuan](https://github.com/search?q=repo%3Ajupyterhub%2Fzero-to-jupyterhub-k8s+involves%3Acbjuan+updated%3A2018-09-03..2019-02-21&type=Issues) | [@cboettig](https://github.com/search?q=repo%3Ajupyterhub%2Fzero-to-jupyterhub-k8s+involves%3Acboettig+updated%3A2018-09-03..2019-02-21&type=Issues) | [@chaoleili](https://github.com/search?q=repo%3Ajupyterhub%2Fzero-to-jupyterhub-k8s+involves%3Achaoleili+updated%3A2018-09-03..2019-02-21&type=Issues) | [@choldgraf](https://github.com/search?q=repo%3Ajupyterhub%2Fzero-to-jupyterhub-k8s+involves%3Acholdgraf+updated%3A2018-09-03..2019-02-21&type=Issues) | [@clancychilds](https://github.com/search?q=repo%3Ajupyterhub%2Fzero-to-jupyterhub-k8s+involves%3Aclancychilds+updated%3A2018-09-03..2019-02-21&type=Issues) | [@clkao](https://github.com/search?q=repo%3Ajupyterhub%2Fzero-to-jupyterhub-k8s+involves%3Aclkao+updated%3A2018-09-03..2019-02-21&type=Issues) | [@consideRatio](https://github.com/search?q=repo%3Ajupyterhub%2Fzero-to-jupyterhub-k8s+involves%3AconsideRatio+updated%3A2018-09-03..2019-02-21&type=Issues) | [@dalonlobo](https://github.com/search?q=repo%3Ajupyterhub%2Fzero-to-jupyterhub-k8s+involves%3Adalonlobo+updated%3A2018-09-03..2019-02-21&type=Issues) | [@danyx23](https://github.com/search?q=repo%3Ajupyterhub%2Fzero-to-jupyterhub-k8s+involves%3Adanyx23+updated%3A2018-09-03..2019-02-21&type=Issues) | [@dhirschfeld](https://github.com/search?q=repo%3Ajupyterhub%2Fzero-to-jupyterhub-k8s+involves%3Adhirschfeld+updated%3A2018-09-03..2019-02-21&type=Issues) | [@djgagne](https://github.com/search?q=repo%3Ajupyterhub%2Fzero-to-jupyterhub-k8s+involves%3Adjgagne+updated%3A2018-09-03..2019-02-21&type=Issues) | [@dkipping](https://github.com/search?q=repo%3Ajupyterhub%2Fzero-to-jupyterhub-k8s+involves%3Adkipping+updated%3A2018-09-03..2019-02-21&type=Issues) | [@easel](https://github.com/search?q=repo%3Ajupyterhub%2Fzero-to-jupyterhub-k8s+involves%3Aeasel+updated%3A2018-09-03..2019-02-21&type=Issues) | [@Edward-liang](https://github.com/search?q=repo%3Ajupyterhub%2Fzero-to-jupyterhub-k8s+involves%3AEdward-liang+updated%3A2018-09-03..2019-02-21&type=Issues) | [@eric-leblouch](https://github.com/search?q=repo%3Ajupyterhub%2Fzero-to-jupyterhub-k8s+involves%3Aeric-leblouch+updated%3A2018-09-03..2019-02-21&type=Issues) | [@evertrol](https://github.com/search?q=repo%3Ajupyterhub%2Fzero-to-jupyterhub-k8s+involves%3Aevertrol+updated%3A2018-09-03..2019-02-21&type=Issues) | [@frouzbeh](https://github.com/search?q=repo%3Ajupyterhub%2Fzero-to-jupyterhub-k8s+involves%3Afrouzbeh+updated%3A2018-09-03..2019-02-21&type=Issues) | [@GergelyKalmar](https://github.com/search?q=repo%3Ajupyterhub%2Fzero-to-jupyterhub-k8s+involves%3AGergelyKalmar+updated%3A2018-09-03..2019-02-21&type=Issues) | [@guoshimin](https://github.com/search?q=repo%3Ajupyterhub%2Fzero-to-jupyterhub-k8s+involves%3Aguoshimin+updated%3A2018-09-03..2019-02-21&type=Issues) | [@h4gen](https://github.com/search?q=repo%3Ajupyterhub%2Fzero-to-jupyterhub-k8s+involves%3Ah4gen+updated%3A2018-09-03..2019-02-21&type=Issues) | [@henchc](https://github.com/search?q=repo%3Ajupyterhub%2Fzero-to-jupyterhub-k8s+involves%3Ahenchc+updated%3A2018-09-03..2019-02-21&type=Issues) | [@hnykda](https://github.com/search?q=repo%3Ajupyterhub%2Fzero-to-jupyterhub-k8s+involves%3Ahnykda+updated%3A2018-09-03..2019-02-21&type=Issues) | [@jgerardsimcock](https://github.com/search?q=repo%3Ajupyterhub%2Fzero-to-jupyterhub-k8s+involves%3Ajgerardsimcock+updated%3A2018-09-03..2019-02-21&type=Issues) | [@joshbode](https://github.com/search?q=repo%3Ajupyterhub%2Fzero-to-jupyterhub-k8s+involves%3Ajoshbode+updated%3A2018-09-03..2019-02-21&type=Issues) | [@jrmlhermitte](https://github.com/search?q=repo%3Ajupyterhub%2Fzero-to-jupyterhub-k8s+involves%3Ajrmlhermitte+updated%3A2018-09-03..2019-02-21&type=Issues) | [@jtlz2](https://github.com/search?q=repo%3Ajupyterhub%2Fzero-to-jupyterhub-k8s+involves%3Ajtlz2+updated%3A2018-09-03..2019-02-21&type=Issues) | [@jzf2101](https://github.com/search?q=repo%3Ajupyterhub%2Fzero-to-jupyterhub-k8s+involves%3Ajzf2101+updated%3A2018-09-03..2019-02-21&type=Issues) | [@kalaytan](https://github.com/search?q=repo%3Ajupyterhub%2Fzero-to-jupyterhub-k8s+involves%3Akalaytan+updated%3A2018-09-03..2019-02-21&type=Issues) | [@lesteve](https://github.com/search?q=repo%3Ajupyterhub%2Fzero-to-jupyterhub-k8s+involves%3Alesteve+updated%3A2018-09-03..2019-02-21&type=Issues) | [@manics](https://github.com/search?q=repo%3Ajupyterhub%2Fzero-to-jupyterhub-k8s+involves%3Amanics+updated%3A2018-09-03..2019-02-21&type=Issues) | [@manycoding](https://github.com/search?q=repo%3Ajupyterhub%2Fzero-to-jupyterhub-k8s+involves%3Amanycoding+updated%3A2018-09-03..2019-02-21&type=Issues) | [@mathematicalmichael](https://github.com/search?q=repo%3Ajupyterhub%2Fzero-to-jupyterhub-k8s+involves%3Amathematicalmichael+updated%3A2018-09-03..2019-02-21&type=Issues) | [@mhuttner](https://github.com/search?q=repo%3Ajupyterhub%2Fzero-to-jupyterhub-k8s+involves%3Amhuttner+updated%3A2018-09-03..2019-02-21&type=Issues) | [@minrk](https://github.com/search?q=repo%3Ajupyterhub%2Fzero-to-jupyterhub-k8s+involves%3Aminrk+updated%3A2018-09-03..2019-02-21&type=Issues) | [@moschlar](https://github.com/search?q=repo%3Ajupyterhub%2Fzero-to-jupyterhub-k8s+involves%3Amoschlar+updated%3A2018-09-03..2019-02-21&type=Issues) | [@nicorikken](https://github.com/search?q=repo%3Ajupyterhub%2Fzero-to-jupyterhub-k8s+involves%3Anicorikken+updated%3A2018-09-03..2019-02-21&type=Issues) | [@nscozzaro](https://github.com/search?q=repo%3Ajupyterhub%2Fzero-to-jupyterhub-k8s+involves%3Anscozzaro+updated%3A2018-09-03..2019-02-21&type=Issues) | [@rahuldave](https://github.com/search?q=repo%3Ajupyterhub%2Fzero-to-jupyterhub-k8s+involves%3Arahuldave+updated%3A2018-09-03..2019-02-21&type=Issues) | [@ryanlovett](https://github.com/search?q=repo%3Ajupyterhub%2Fzero-to-jupyterhub-k8s+involves%3Aryanlovett+updated%3A2018-09-03..2019-02-21&type=Issues) | [@satra](https://github.com/search?q=repo%3Ajupyterhub%2Fzero-to-jupyterhub-k8s+involves%3Asatra+updated%3A2018-09-03..2019-02-21&type=Issues) | [@scivm](https://github.com/search?q=repo%3Ajupyterhub%2Fzero-to-jupyterhub-k8s+involves%3Ascivm+updated%3A2018-09-03..2019-02-21&type=Issues) | [@sgloutnikov](https://github.com/search?q=repo%3Ajupyterhub%2Fzero-to-jupyterhub-k8s+involves%3Asgloutnikov+updated%3A2018-09-03..2019-02-21&type=Issues) | [@stefansedich](https://github.com/search?q=repo%3Ajupyterhub%2Fzero-to-jupyterhub-k8s+involves%3Astefansedich+updated%3A2018-09-03..2019-02-21&type=Issues) | [@stevenstetzler](https://github.com/search?q=repo%3Ajupyterhub%2Fzero-to-jupyterhub-k8s+involves%3Astevenstetzler+updated%3A2018-09-03..2019-02-21&type=Issues) | [@tianhuil](https://github.com/search?q=repo%3Ajupyterhub%2Fzero-to-jupyterhub-k8s+involves%3Atianhuil+updated%3A2018-09-03..2019-02-21&type=Issues) | [@TiemenSch](https://github.com/search?q=repo%3Ajupyterhub%2Fzero-to-jupyterhub-k8s+involves%3ATiemenSch+updated%3A2018-09-03..2019-02-21&type=Issues) | [@tjcrone](https://github.com/search?q=repo%3Ajupyterhub%2Fzero-to-jupyterhub-k8s+involves%3Atjcrone+updated%3A2018-09-03..2019-02-21&type=Issues) | [@tmshn](https://github.com/search?q=repo%3Ajupyterhub%2Fzero-to-jupyterhub-k8s+involves%3Atmshn+updated%3A2018-09-03..2019-02-21&type=Issues) | [@trallard](https://github.com/search?q=repo%3Ajupyterhub%2Fzero-to-jupyterhub-k8s+involves%3Atrallard+updated%3A2018-09-03..2019-02-21&type=Issues) | [@vilhelmen](https://github.com/search?q=repo%3Ajupyterhub%2Fzero-to-jupyterhub-k8s+involves%3Avilhelmen+updated%3A2018-09-03..2019-02-21&type=Issues) | [@willingc](https://github.com/search?q=repo%3Ajupyterhub%2Fzero-to-jupyterhub-k8s+involves%3Awillingc+updated%3A2018-09-03..2019-02-21&type=Issues) | [@wydwww](https://github.com/search?q=repo%3Ajupyterhub%2Fzero-to-jupyterhub-k8s+involves%3Awydwww+updated%3A2018-09-03..2019-02-21&type=Issues) | [@yugushihuang](https://github.com/search?q=repo%3Ajupyterhub%2Fzero-to-jupyterhub-k8s+involves%3Ayugushihuang+updated%3A2018-09-03..2019-02-21&type=Issues) | [@yuvipanda](https://github.com/search?q=repo%3Ajupyterhub%2Fzero-to-jupyterhub-k8s+involves%3Ayuvipanda+updated%3A2018-09-03..2019-02-21&type=Issues) | [@zonca](https://github.com/search?q=repo%3Ajupyterhub%2Fzero-to-jupyterhub-k8s+involves%3Azonca+updated%3A2018-09-03..2019-02-21&type=Issues) | [@zxcGrace](https://github.com/search?q=repo%3Ajupyterhub%2Fzero-to-jupyterhub-k8s+involves%3AzxcGrace+updated%3A2018-09-03..2019-02-21&type=Issues)

**Comment by consideRatio** [comment]
What about the following.

1. We use github-activity's definition of contribution, and to collect contributors.
2. We attribute everyone in z2jh part of its version bump ...
3. ... and recursively attribute every contributor to a Jupyter ecosystem project part of a version bump in z2jh's version bump.
4. We let step 3 to be manual when it comes to inspecting versions bumped. This involves inspecting image tags for configurable-http-proxy, and the various installed versions in the hub image. Note that currently we only attribute for z2jh/jupyterhub/kubespawner/oauthenticator, excluding other authenticators, configurable-http-proxy, and all non-jupyter ecosystem projects.
5. We develop a way to combine the various subprojects contributors to a final list.
6. We ensure that the entire process about attribution is likely to be doable in less than 30 minutes for someone reading release documentation for the first time.

**Comment by yuvipanda** [comment]
@consideRatio I LOVE IT! \o/ <3. I  appreciate your efforts to consistently make processes better for everyone involved.

**Comment by choldgraf** [comment]
this sounds great to me!

(also just a note that `github-activity` will work with *just* a github organization, at which point it list contributors across *all repositories in the org* within a window of time)

**Comment by consideRatio** [comment]
@yuvipanda @choldgraf I have released 0.10.0-beta.1 of the Helm chart, but I just used github-activity straight up. I'm feeling it is too much work to manually combine oauthenticator / kubespawner / jupyterhub / configurable-http-proxy contributions with github-activity or use the contributions.py script which have taken multiple 30 minute attempts in sequence with pauses for rate-limiting in between as well as a manual process of deciding on exactly what versions and dates etc to use for various repos.

I'd like to reduce the maintenance burden in general for JupyterHub org repos, so I'd suggest that a distribution like this highlights that this is an open source project built from other important pieces of open source software within the JupyterHub org, and link to those projects changelogs with their dedicated attribution etc. We could do that next to a version map where we can highlight how thie z2jh release bumped KubeSpawner from version X to version Y for example.

Hmmm...

**Comment by choldgraf** [comment]
If this is a significant maintainer burden to run these scripts, I think that we should find alternatives. I like the approach that you mention @consideRatio - to ensure that we follow best-practices for thanking contributors in sub-projects, and then make sure we link to those projects from the more central ones like z2jh. Perhaps more infrequently (1 / 2 times a year?) we can write an aggregate "thank you" post that aggregates across all of the repositories in the jupyterhub org. WDYT?

**Comment by consideRatio** [comment]
> Perhaps more infrequently (1 / 2 times a year?) we can write an aggregate "thank you" post that aggregates across all of the repositories in the jupyterhub org. WDYT?

Ohhh perhaps we can have a appreciation day, like a birth day for JupyterHub org contributors, perhaps twice a year? :)

**Comment by consideRatio** [comment]
@yuvipanda @choldgraf I have updated the RELEASE.md file as part of this PR, which now includes a bullet point about celebrating projects z2jh depends on to be fleshed out a bit soon as part of the final 0.10.0 release. In other words, I aim to follow this idea...

> a distribution like this highlights that this is an open source project built from other important pieces of open source software within the JupyterHub org, and link to those projects changelogs with their dedicated attribution etc.

---

I suggest that we merge this at this point and accept the changes in the release section as they are, I have validated that they are reasonable for me at least as part of #1880.

File: RELEASE.md [hunk 1]
```diff
@@ -6,125 +6,90 @@ coordinate. Do some copy pasting!
...
+This issue will be used to coordinate the next release of the Helm chart according to the instructions in `RELEASE.md`. Below is a checklist for this release.
 
 ## Look through dependencies
 
-The JupyterHub Helm chart relies on many dependent projects, and when we make a
-release it is good to be updated about their status and what version we decide
-to couple the Helm chart release with. Below are the more important depdencies.
-Put a check on those that reach a state good enough for a z2jh release to be
-cut.
+The JupyterHub Helm chart relies on many dependent projects, and when we make a release it is good to be updated about their status and what version we decide to couple the Helm chart release with. Below are the more important depdencies. Put a check on those that reach a state good enough for a z2jh release to be cut.
```
Comment (choldgraf) [inline RELEASE.md:13]:
```suggestion
The JupyterHub Helm chart relies on many dependent projects, and when we make a release we should note major updates for these projects as well as the version that is specified in the Helm chart. Below are the more important dependencies. Put a check on those that reach a state good enough for a z2jh release to be cut.
```

Comment (choldgraf) [reply RELEASE.md:13]: I'm not quite sure what "state good enough for a z2jh release" means...do you have an idea for what that is getting at?

**Review by choldgraf — COMMENTED** [review]
In general this looks great - I'm glad that the new instructions seem shorter and less-involved :-)

I had one quick comment in there but IMO it's good to go

**Comment by consideRatio** [comment]
This is the outcome of my work to improve the attribution to other projects so far in #1886.

image

image

@yuvipanda are you okay to sunset the contributions script?

**Comment by choldgraf** [comment]
I love that table 👍

**Comment by yuvipanda** [comment]
This is awesome work, @consideRatio! <3 thank you :)

**Pull request merged**